### ISSUE TITLE
SFMLOrthogonolLayer.hpp Update() Animation Error

### ISSUE THREAD

**Issue by devchild** [fallahn/tmxlite#105]
Hi..
The example on SFML does not handle the animation correctly... please find a more correct implementation of the update function in the SFMLOrthogonolLayer.hpp

```
    void update(sf::Time elapsed) {
        for (auto& c : m_visibleChunks) {
            for (AnimationState& as : c->getActiveAnimations()) {
                as.currentTime += elapsed;

                tmx::TileLayer::Tile tile;
                std::uint32_t animTime = 0;
                auto x = as.animTile.animation.frames.begin();
                while (animTime < as.currentTime.asMilliseconds()) {
                    if (x == as.animTile.animation.frames.end()) {
                        x = as.animTile.animation.frames.begin();
                        as.currentTime -= sf::milliseconds(animTime);
                        animTime = 0;
                    }

                    tile.ID = x->tileID;
                    animTime += x->duration;
                    x++;
                }

                setTile(as.tileCords.x, as.tileCords.y, tile);
            }
        }
    }
```

**Comment by fallahn** [comment]
Hi, thanks for the feedback! Could you explain in a little more detail what this does differently? Thanks

**Comment by devchild** [comment]
When I run the example using the original code, I have the impression that the tile-frame gets changed, even when the elapsed time isn't larger than the frames duration...Also when the animation needs to be reset, there is still residual elapsed time, this should also be taken into account in the AnimationState.

**Comment by fallahn** [comment]
I see what you mean - if the residual time isn't carried over then the next frame time is going to be incorrect. If you'd like to submit a pull request (and move the opening braces to a new line to match the existing style, purely for consistency's sake) I'd be grateful, thanks!

**Comment by fallahn** [comment]
Closed by #106
